script.js

```diff
@@ -1,4 +1,238 @@
+
+  //Winning conditions
+  if(
+    (userPlays == `scissors` && computerPlays == `paper`)||
+    (userPlays == `paper` && computerPlays == `stone`) ||
+    (userPlays == `stone` && computerPlays == `scissors`)
+  ){
+    numTimesUserWins += 1;
+    return `${gamePlayMessage}<br><br>${userName} wins! Yay!`;
+  }else{
+    //Draw conditions
+    if(userPlays == computerPlays){
+        numTimesDraw += 1;
+        return `${gamePlayMessage}<br><br>It's a draw.`;
+      }else{
+        //if not, assume computer wins
+        numTimesComputerWins += 1;
+        return `${gamePlayMessage}<br><br>${userName} loses.`
+      };
+  };
+};
+
+//REVERSE GAME LOGIC AND SCORING
+var reverseGameLogic = function(userPlays,computerPlays){
+  var gamePlayMessage = `Computer chose ${createEmojiSPS(computerPlays)}.<br>
+  ${userName} chose ${createEmojiSPS(userPlays)}.`
+
+  //Winning conditions
+  if(
+    (userPlays == `scissors` && computerPlays == `stone`) ||
+    (userPlays == `paper` && computerPlays == `scissors`) ||
+    (userPlays == `stone` && computerPlays == `paper`)
+  ){
+    numTimesUserWins += 1;
+    return `${gamePlayMessage}<br><br>${userName} wins! Yay!`;
+  }else{
+    //Draw conditions
+    if(userPlays == computerPlays){
+        numTimesDraw += 1;
+        return `${gamePlayMessage}<br><br>It's a draw.`;
+      }else{
+        //if not, assume computer wins
+        numTimesComputerWins += 1;
+        return `${gamePlayMessage}<br><br>${userName} loses.`;
+    };
+  };  
+};
+
+//KOREAN GAME LOGIC AND SCORING
+var koreanGameLogic = function(userPlays,computerPlays){
+  //Most recent winner conditions
+  if(
+    (userPlays == `scissors` && computerPlays == `paper`)||
+    (userPlays == `paper` && computerPlays == `stone`) ||
+    (userPlays == `stone` && computerPlays == `scissors`)
+  ){
+    mostRecentWinner = userName;
+    numTimesUserWins += 1;
+    return `${userName} won this round. It is ${userName}'s turn to attack.`
+  }else{
+      if(
+        (computerPlays == `scissors` && userPlays == `paper`)||
+        (computerPlays == `paper` && userPlays == `stone`) ||
+        (computerPlays == `stone` && userPlays == `scissors`)
+      ){
+        mostRecentWinner = `Computer`;
+        numTimesComputerWins += 1;
+        return `Computer won this round. It is Computer's turn to attack.`
+      }else{
+        //Ultimate winner conditions
+          if(userPlays == computerPlays){
+            if(mostRecentWinner == userName){
+              mostRecentWinner = ``;
+              return `${userName} wins the game!`
+            }else{
+                  if(mostRecentWinner == `Computer`){
+                    mostRecentWinner = ``;
+                    return `Computer wins the game!`
+                  }else{
+                      numTimesDraw += 1;
+                      return `No attacker yet. Keep playing`;
+                  };
+            };
+          };
+      };
+  };
+};
+
+//generate random number from 1 to 3
```

Comment:
these 3 functions should go at the top of your code because functions should be defined before they are called